### PR TITLE
add the option for disabling utf8 en/de-coding

### PR DESCRIPTION
**Pull request by shogo82148**
fix https://github.com/plu/Pithub/issues/187

### CODE REVIEW

File: lib/Pithub/Base.pm [hunk 1]
```diff
@@ -727,6 +732,17 @@ sub _build__json {
...
+    my $self = shift;
+    $self->_json->utf8(@_);
+    return $self;
+}
+
+sub get_utf8 {
+    my $self = shift;
+    return $self->_json->get_utf8(@_);
+}
+
```
Comment (oalders) [inline lib/Pithub/Base.pm]: Is there any reason why you haven't set this up as a Moo attribute?

```
has utf8 => (
    is      => 'ro',
    default => 1,
);
```

File: t/encoding.t [hunk 2]
```diff
@@ -11,12 +11,14 @@ BEGIN {
 
 {
     my $obj = Pithub::Test::Factory->create('Pithub::Users');
+    ok $obj->get_utf8, 'enabled en/de-conding json';
```
Comment (oalders) [inline t/encoding.t]: I think this should be `enabled en/decoding json`.

**Comment by oalders** [comment]
This looks good to me.  I think it would make sense to use a Moo attribute, as I've commented, but otherwise I think we can ship this.  Anyone else care to weigh in?  @shogo82148 I apologize for the delay.  I haven't been feeling too well for a few weeks.

File: t/encoding.t (hunk 2 above)
Comment (shogo82148) [inline t/encoding.t]: I will fix it

File: lib/Pithub/Base.pm (hunk 1 above)
Comment (shogo82148) [inline lib/Pithub/Base.pm]: It is same interface with JSON.pm.
I agree to use Moo attribute.

**Comment by shogo82148** [comment]
I set the utf8 option as a Moo attribute and fixed spelling.

**Comment by oalders** [comment]
:+1: Thank you!

**Comment by oalders** [comment]
These changes have just been shipped to CPAN.

**Comment by shogo82148** [comment]
Thank you!

**Pull request merged**